From: Alex Crichton <alex@alexcrichton.com>
Date: Mon, 22 Jul 2024 07:57:53 -0700
Subject: Fix inclusion of `wasm-component-ld` in dist artifacts

This is another accidental omission from #126967 (in addition
to #127867) which fixes an issue where `wasm-component-ld` isn't
distributed via rustup just yet because while it's present in the
sysroot it's not present in the tarballs.

(cherry picked from commit c3d3d6fe0295c2521801fc4d0c6afe73e4d1328c)

Forwarded: https://github.com/rust-lang/rust/pull/128060
---
 src/bootstrap/src/core/build_steps/dist.rs | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/bootstrap/src/core/build_steps/dist.rs b/src/bootstrap/src/core/build_steps/dist.rs
index 550d88c..a91361e 100644
--- a/src/bootstrap/src/core/build_steps/dist.rs
+++ b/src/bootstrap/src/core/build_steps/dist.rs
@@ -473,6 +473,11 @@ impl Step for Rustc {
                     );
                 }
             }
+            if builder.build_wasm_component_ld() {
+                let src_dir = builder.sysroot_libdir(compiler, host).parent().unwrap().join("bin");
+                let ld = exe("wasm-component-ld", compiler.host);
+                builder.copy_link(&src_dir.join(&ld), &dst_dir.join(&ld));
+            }
 
             // Man pages
             t!(fs::create_dir_all(image.join("share/man/man1")));
